recorder/src/main/cpp/controller.cc

```diff
@@ -36,7 +37,13 @@ void Controller::start() {
 }
 
 void Controller::stop() {
-    keep_running.store(false, std::memory_order_relaxed);
+    cancel_work();
+    scheduler.schedule(Time::now(), [&]() {
```

Comment:
Later in issue_work, we also schedule 1 task to mark the queue read-only and retire all work. Is the order of these scheduled task deterministic. To me it looks like keep_running will be marked false first.